Exdi/exdigdbsrvsample/Start-ExdiDebugger.ps1

```diff
@@ -0,0 +1,314 @@
+        $ExdiInstallDir = [System.IO.Path]::GetDirectoryName($(Get-ExdiInstallPath))
+        if (-not (Test-Path $ExdiInstallDir))
+        {
+            Write-Host "Currently Registered exdi server does not exist. Reinstalling..."
+            return $false
+        }
+        elseif (-not ((Test-Path "$ExdiInstallDir\exdiConfigData.xml") -and (Test-Path "$ExdiInstallDir\systemregisters.xml")))
+        {
+            Write-Host "Currently Registered exdi server does not have required xml settings files. Reinstalling..."
+            return $false
+        }
+        else
+        {
+            Write-Verbose "Exdi server is insalled correctly. Skipping installation..."
+            return $true
+        }
+    }
+    else
+    {
+        Write-Host "Exdi server is not installed. Installing..."
+        return $false
+    }
+}
+
+Function Install-ExdiServer
+{
+    [CmdletBinding()]
+    param
+    (
+        [string] $InstallFrom,
+        [string] $InstallTo
+    )
+    
+    if (-not $(Test-Admin))
+    {
+        throw "Script needs to be run as an Admin to install exdi software."
+    }
+
+    New-Item -ItemType Directory $InstallTo -ErrorAction SilentlyContinue | Write-Verbose
+    Copy-Item -Path "$InstallFrom\ExdiGdbSrvSample.dll" -Destination $InstallTo -ErrorAction stop | Write-Verbose
+    Copy-Item -Path "$InstallFrom\exdiConfigData.xml" -Destination $InstallTo -ErrorAction stop | Write-Verbose
+    Copy-Item -Path "$InstallFrom\systemregisters.xml" -Destination $InstallTo -ErrorAction stop | Write-Verbose
+    regsvr32 /s "$InstallTo\ExdiGdbSrvSample.dll"
+
+    if ($(Get-ExdiInstallPath) -eq $null)
+    {
+        throw "Unable to install exdi server"
+    }
+}
+
+Function Edit-ExdiConfigFile
+{
+    [CmdletBinding()]
+    param
+    (
+        [string] $ExdiFilePath,
+        [string] $ExdiTargetType,
+        [PSCustomObject[]] $XmlSettingPathValueList
+    )
+    
+    # Edit exdiConfigData.xml
+    [xml]$exdiConfigXml = Get-Content "$ExdiFilePath"
+
+    # Set current target
+    $exdiConfigXml.ExdiTargets.CurrentTarget = $ExdiTarget
+
+    # set HostNameAndPort
+    $ExdiTargetXmlNode = $exdiConfigXml.SelectSingleNode("//ExdiTargets/ExdiTarget[@Name='$ExdiTarget']/ExdiGdbServerConfigData")
+
+    foreach ($XmlSettingPathValue in $XmlSettingPathValueList)
+    {
+        Write-Verbose "Processing $XmlSettingPathValue"
+        if ($XmlSettingPathValue.Value -eq $null)
+        {
+            continue
+        }
+
+        $PathParts = $XmlSettingPathValue.Path.Split(".")
+        $curNode = $ExdiTargetXmlNode
+        if ($PathParts.Count -gt 1)
+        {
+            foreach ($PathPart in $PathParts[0..($PathParts.Count-2)])
+            {
+                Write-Verbose $PathPart
+                $curNode = $curNode.($PathPart)
+            }
+        }
+        $curNode.($PathParts[-1]) = $XmlSettingPathValue.Value
+    }
+
+    $exdiConfigXml.Save("$ExdiFilePath")
+}
+
+Function Stop-ExdiContainingDllHosts
```

Comment:
I'm a bit surprised that EXDI doesn't shut itself down when windbg closes. An argument in favor of moving the component in-proc... but it still seems like something that should be fixable even out-of-proc.

Reply:
Agreed. This should be investigated at some point...